python/kubeflow/trainer/utils/utils.py

```diff
@@ -107,6 +108,74 @@ def get_runtime_trainer_container(
     return None
 
 
+def _detect_trainer_from_image_patterns(image_name: str) -> Optional[types.Trainer]:
+    """
+    Detect trainer type based on image name patterns using regex.
+
+    This method uses pattern matching on the image name to determine
+    the likely trainer type.
+
+    Args:
+        image_name: The container image name
+
+    Returns:
+        Trainer object if detected, None otherwise
+    """
+    # DeepSpeed patterns
+    if re.search(r"deepspeed", image_name, re.IGNORECASE):
+        return copy.deepcopy(types.TRAINER_CONFIGS[types.Framework.DEEPSPEED])
+
+    # MLX patterns
+    if re.search(r"mlx", image_name, re.IGNORECASE):
+        return copy.deepcopy(types.TRAINER_CONFIGS[types.Framework.MLX])
+
+    # TorchTune patterns (check before PyTorch to avoid conflicts)
+    if re.search(r"torchtune", image_name, re.IGNORECASE):
+        return copy.deepcopy(types.TRAINER_CONFIGS[types.Framework.TORCHTUNE])
+
+    # PyTorch patterns (more specific to avoid matching torchtune)
+    if re.search(r"pytorch", image_name, re.IGNORECASE):
+        return copy.deepcopy(types.TRAINER_CONFIGS[types.Framework.TORCH])
+
+    # Generic torch patterns (but not torchtune)
+    if re.search(r"^torch(?!tune)", image_name, re.IGNORECASE):
+        return copy.deepcopy(types.TRAINER_CONFIGS[types.Framework.TORCH])
```

Comment:
I think, we can agree that PyTorch based runtime must contain `pytorch` in the image name.
In that case, we can distinguish `torchtune` and `pytorch` images.
```suggestion
```